src/timestamp.rs

```diff
@@ -47,7 +41,7 @@ impl Display for Timestamp {
         match self {
             Self::DDHHMM(d, h, m) => write!(f, "{:02}{:02}{:02}z", d, h, m),
             Self::HHMMSS(h, m, s) => write!(f, "{:02}{:02}{:02}h", h, m, s),
-            Self::Unsupported(s) => write!(f, "{}", s),
+            Self::Unsupported(s) => write!(f, "{}", String::from_utf8_lossy(s)),
```

Comment:
Same issue here as with `AprsMessage::Display`. This will not round-trip timestamps containing invalid unicode correctly, and it's being used by `AprsPosition::encode`, which afaui is expected to round-trip correctly.